sourcegraph/eclipse
# Add log view

This adds unified log of events (logged messages, jsonrpc messages, webview exceptions) and view to present them.

## Test plan
- Open Cody Log view (for example `Alt + Shift + Q, Q` and select `Cody Log`)
- The new view will contain environment information (plugin, jvm & os versions) and below that events such as jsonrpc messages.
- Write something in the `filter` textbox on the right side of the view, the events should be filtered (It only supports basic substring matching).
- Check and uncheck boxes next to the event types below the text field. It will filter events by type.
- Click `copy` action button. Paste logs somewhere.
- Click `clear` button. All events will be removed. The environment information will stay.
- Check if log calls are forwarded to the eclipse console. If no errors occurred it should contain two messages with info level, about starting the webserver and the agent.
- Put some `throw` statement in the javascript resource and restart eclipse. The error will be logged.



[Screencast from 13.08.2024 16:57:36.webm](https://github.com/user-attachments/assets/e3b9c967-c13d-43c1-bdc3-0f2d1de0a3cc)
